### PR TITLE
 fix #17484: 修复tsx文件对象层自定义属性读取错误 

### LINKED ISSUE #17484: Tiled文件对象层属性读取错误
### Cocos Creator version

3.8.4

### System information

Windows10

### Issue description

[问题 1](https://github.com/cocos/cocos-engine/blob/fc7c067d14c1c2a452ea064bee2ea3d5497afc73/cocos/tiledmap/tmx-xml-parser.ts#L923)

```xml
<objectgroup id="2" name="Object">
   <properties>
     <property name="Start" type="bool" value="false"/>
   </properties>
   <object id="32" name="SavePoint" x="480" y="2040" width="120" height="120">
     <properties>
       <property name="Start" type="bool" value="true"/>
     </properties>
   </object>
</objectgroup>
```

在读取属性 property 时调用 `getElementsByTagName` 会得到所有后代节点中的匹配元素。因此，在读取 objectgroup 的 properties 节点时，如果子节点 object 也有 properties 节点，则会将 object 的 propertie 覆盖到 objectgroup 上，建议改为`querySelectorAll(':scope > properties')`。

[问题 2](https://github.com/cocos/cocos-engine/blob/fc7c067d14c1c2a452ea064bee2ea3d5497afc73/cocos/tiledmap/tmx-xml-parser.ts#L948C17-L948C32)
在读取 object 的 properties 时，会直接将 property 的值覆盖到 object 上。应将这些属性值存储在 properties 字段中，和其他地方保持一致。

### Relevant error log output

_No response_

### Steps to reproduce

_No response_

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by Jerry457**
## Greptile Summary

This pull request addresses issue #17484 by fixing the error in reading custom properties of object layers from TSX files.

- Added `properties` field to `TMXObject` interface in `cocos/tiledmap/tiled-types.ts`.
- Updated `getPropertyList` function in `cocos/tiledmap/tmx-xml-parser.ts` to use `querySelectorAll` for better property selection.
- Modified `parseObjectGroup` method in `cocos/tiledmap/tmx-xml-parser.ts` to correctly assign properties to object layers.

### CODE REVIEW

File: cocos/tiledmap/tmx-xml-parser.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ function strToColor (value: string): Color {
 
 function getPropertyList (node: Element, map?: PropertiesInfo): PropertiesInfo {
     const res: any[] = [];
-    const properties = node.getElementsByTagName('properties');
+    const properties = node.querySelectorAll(':scope > properties');
```
Comment (qiuguohua) [inline cocos/tiledmap/tmx-xml-parser.ts]: WeChat mini-games do not support this interface

Comment (qiuguohua) [reply cocos/tiledmap/tmx-xml-parser.ts]: @Jerry457

Comment (Jerry457) [reply cocos/tiledmap/tmx-xml-parser.ts]: fixed

Comment (qiuguohua) [reply cocos/tiledmap/tmx-xml-parser.ts]: @Jerry457 Please fix test error.

Comment (qiuguohua) [reply cocos/tiledmap/tmx-xml-parser.ts]: And modify the branch to 3.8.5

**Comment by qiuguohua** [comment]
@Jerry457 There is also a ci error

File: cocos/tiledmap/tmx-xml-parser.ts (hunk 1 above)
Comment (Jerry457) [reply cocos/tiledmap/tmx-xml-parser.ts]: Fixed in new PR: https://github.com/cocos/cocos-engine/pull/17641

**Pull request closed without merging**